Formula/d/dwarfs.rb

```diff
@@ -0,0 +1,167 @@
+class Dwarfs < Formula
+  desc "Fast high compression read-only file system for Linux, Windows, and macOS"
+  homepage "https://github.com/mhx/dwarfs"
+  url "https://github.com/mhx/dwarfs/releases/download/v0.10.1/dwarfs-0.10.1.tar.xz"
+  sha256 "db785e0e0f257fa4363d90153db34127add4552791a72998b30ded787840d039"
+  license "GPL-3.0-or-later"
+
+  # At least Apple Clang 15 is required as we need several C++20 features.
+  NEED_LLVM_CLANG = (OS.mac? && DevelopmentTools.clang_build_version < 1500).freeze
+
+  depends_on "cmake" => :build
+  depends_on "googletest" => :build
+  depends_on "pkg-config" => :build
+  depends_on "boost"
+  depends_on "brotli"
+  depends_on "double-conversion"
+  depends_on "flac"
+  depends_on "fmt"
+  depends_on "gflags"
+  depends_on "glog"
+  depends_on "howard-hinnant-date"
+  depends_on "libarchive"
+  depends_on "libevent"
+  depends_on "libsodium"
+  depends_on "llvm" if NEED_LLVM_CLANG
+  depends_on "lz4"
+  depends_on "nlohmann-json"
+  depends_on "openssl@3"
+  depends_on "parallel-hashmap"
+  depends_on "range-v3"
+  depends_on "utf8cpp"
+  depends_on "xxhash"
+  depends_on "xz"
+  depends_on "zstd"
+
+  on_linux do
+    depends_on "jemalloc"
+  end
+
+  fails_with :clang do
+    build 1400
+    cause "Not all required C++20 features are supported"
+  end
+
+  fails_with :gcc do
+    version "11"
+    cause "Not all required C++20 features are supported"
+  end
+
+  def install
+    args = std_cmake_args + %W[
+      -DBUILD_SHARED_LIBS=ON
+      -DCMAKE_PREFIX_PATH=#{libexec}
+      -DCMAKE_INSTALL_RPATH=#{rpath}
+      -DWITH_LIBDWARFS=ON
+      -DWITH_TOOLS=ON
+      -DWITH_FUSE_DRIVER=OFF
+      -DWITH_TESTS=ON
+      -DWITH_MAN_PAGES=ON
+      -DENABLE_PERFMON=ON
+      -DTRY_ENABLE_FLAC=ON
+      -DENABLE_RICEPP=ON
+      -DENABLE_STACKTRACE=OFF
+      -DDISABLE_MOLD=ON
+      -DPREFER_SYSTEM_GTEST=ON
+    ]
+
+    ctest_args = %w[
+      --output-on-failure
+    ]
+
+    if NEED_LLVM_CLANG
+      ENV.llvm_clang
+
+      # Needed in order to find the C++ standard library
+      # See: https://github.com/Homebrew/homebrew-core/issues/178435
+      ENV.prepend "LDFLAGS", "-L#{Formula["llvm"].opt_lib}/c++ -L#{Formula["llvm"].opt_lib}"
+
+      # Unfortunately, even with the above *and* explicitly adding `-lunwind`,
+      # boost::wrapexcept still doesn't get caught. This is really only an issue
+      # for parsing command line arguments, so we exclude the related tests here.
+      # See: https://github.com/Homebrew/homebrew-core/issues/175994
+      ctest_args += %w[
+        --exclude-regex .*main_test.invalid_cmdline_args
+      ]
+    end
+
+    system "cmake", "--fresh", "-S", ".", "-B", "build", *args
+    system "cmake", "--build", "build", "--parallel", ENV.make_jobs
+    system "ctest", "--test-dir", "build", "--parallel", ENV.make_jobs, *ctest_args
```

Comment:
We don't run tests unless it's a critical encryption library
```suggestion
```

Reply:
Just curious, what's the reason behind this?

When building from source, the tests contribute only a small fraction in terms of run-time.
When the user is installing a bottled release, the tests won't run anyway.
When actually building the bottles, I'd feel more comfortable if they were tested.

Sure, there's the `test` section, but that only superficially checks if the stuff is horribly broken.

Reply:
Because it's up to upstream to do unit tests, all Homebrew wants to know is "did we install it right" which won't be detected by a step that runs before the install.

Reply:
That would require upstream to have access to all platforms that Homebrew supports. Considering that's 3 versions of macOS across two architectures, plus Linux, I'd be surprised if even just a small fraction of projects with Homebrew formulae could pull that off.

Reply:
A quick grep finds this:

https://github.com/Homebrew/homebrew-core/blob/master/Formula/q/quick-lint-js.rb#L41

This is hardly a "critical encryption library". ;)